File: HW/HW2/findBadVersion.py
def first_bad_version(n):
        left, right = 1, n

        while left < right:
            mid = left + (right - left) // 2
            if VersionControl.isBadVersion(mid):
                right = mid
            else:
                left = mid + 1

        return left

n = 10  # Assuming there are 10 versions
print("The first bad version is:", 

File: HW/HW2/findBadVersion.java
public class findBadVersion {
	public static int firstBadVersion(int n) {
		int left = 1;
		int right = n;
		
		while (left < right) {
			int mid = left + (right - left)/2;
			if (isBadVersion(mid)) {
				right = mid;
			} else {
				left = mid +1; 
			}
		}
		return left;
	}
    public static void main(String[] args) {
        // Test your function with sample inputs
        int n = 10;
        System.out.println("First bad version is: "+ firstBadVersion(n));
    }
}
